File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepageFragment.java
package com.wayfair.themoviedb.feature.homepage;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.SearchView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.pubbix.base.BaseFragment;
import com.wayfair.themoviedb.R;

import org.jetbrains.annotations.NotNull;

import java.util.UUID;

public class HomepageFragment extends BaseFragment<HomepagePresenter> implements HomepageContract.View {

    private SearchView searchView;

    public static Fragment newInstance() {
        Bundle bundle = new Bundle();
        bundle.putString("instance_id", UUID.randomUUID().toString());
        Fragment fragment = new HomepageFragment();
        fragment.setArguments(bundle);
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        View view = inflater.inflate(R.layout.fragment_homepage, container, false);
        searchView = view.findViewById(R.id.search_view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        presenter.setUpViews();
    }

    @Override
    public void showMessage(@NotNull String message) {

    }

    @Override
    public void setOnQueryTextListener() {
        searchView.setOnQueryTextListener(new SearchView.OnQueryTextListener() {
            @Override
            public boolean onQueryTextSubmit(String s) {
                //TODO Needs to add check here
                presenter.search(s);
                return true;
            }

            @Override
            public boolean onQueryTextChange(String s) {
                return false;
            }
        });
    }
}

File: app/src/main/java/com/wayfair/themoviedb/feature/homepage/HomepageFragment.kt
package com.wayfair.themoviedb.feature.homepage

import android.content.Context
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.SearchView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.pubbix.base.BaseFragment
import com.wayfair.themoviedb.R
import kotlinx.android.synthetic.main.fragment_homepage.*
import java.util.*

class HomepageFragment : BaseFragment<HomepagePresenter>(), HomepageContract.View {

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        return inflater.inflate(R.layout.fragment_homepage, container, false)
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        presenter.attachView(this)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        search_view.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
            override fun onQueryTextSubmit(s: String): Boolean {
                presenter.search(s)
                return true
            }

            override fun onQueryTextChange(s: String): Boolean {
                return false
            }
        })
    }

    override fun onDetach() {
        super.onDetach()
        presenter.detachView()
    }

    override fun showMessage(message: String) {

    }

    companion object {
        fun newInstance(): Fragment {
            val bundle = Bundle()
            bundle.putString("instance_id", UUID.randomUUID().toString())
            val fragment = HomepageFragment()
            fragment.arguments = bundle
            return fragment
        }
    }
}
